### ISSUE TITLE
Report error: object keys cannot be template literals

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#124]
The following code is illegal:

```js
let o = {`key`: value};
```

Report a nice diagnostic for this issue.

**Comment by singalhimanshu** [comment]
Can you please point me to what files should I look at?

**Comment by strager** [comment]
> Can you please point me to what files should I look at?

I think test-parse-expression.cpp and parse.cpp. In the latter, there should be a function called something like `parse_object_literal`.

**Comment by strager** [comment]
@singalhimanshu How is progress on this task? Do you need any assistance?

**Comment by alexanderkarlis** [comment]
@singalhimanshu Think you said last night it would be alright if I gave this one a crack? Thanks!
